library/cc/string_accessor.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+#include "absl/types/optional.h"
+#include "library/common/api/c_types.h"
+
+namespace Envoy {
+namespace Platform {
+
+/**
+ * `StringAccessor` is an interface that may be implemented to provide access to an arbitrary
+ * string from the application.
+ */
+class StringAccessor : public std::enable_shared_from_this<StringAccessor> {
+public:
+  virtual ~StringAccessor() = default;
+
+  /**
+   * Returns the string associated with this accessor.
+   */
+  virtual const std::string& getString() const PURE;
```

Comment:
maybe let's just call it `get()`? `String` part is clear based on the signature of the method. 

No strong opinion

Reply:
Done. (I'd considered this initially but the Java impl had string in the name. But if you and I both prefer simple "get()", then I'm sold :>)